pkg/apis/acme/v1alpha2/types_issuer.go

```diff
@@ -479,19 +479,31 @@ type ACMEIssuerDNS01ProviderAzureDNS struct {
 	// +optional
 	ClientSecret *cmmeta.SecretKeySelector `json:"clientSecretSecretRef,omitempty"`
 
+	// ID of the Azure subscription
 	SubscriptionID string `json:"subscriptionID"`
 
 	// when specifying ClientID and ClientSecret then this field is also needed
 	// +optional
 	TenantID string `json:"tenantID,omitempty"`
 
+	// resource group the DNS zone is located in
 	ResourceGroupName string `json:"resourceGroupName"`
 
+	// name of the DNS zone that should be used
 	// +optional
 	HostedZoneName string `json:"hostedZoneName,omitempty"`
 
+	// name of the Azure environment (default AzurePublicCloud)
 	// +optional
 	Environment AzureDNSEnvironment `json:"environment,omitempty"`
+
+	// clientID of the MSI that should be used (only relevant when using MSI, can not be used at the same time as ManagedIdentityResourceID)
+	// +optional
+	ManagedIdentityClientID string `json:"managedIdentityClientID,omitempty"`
+
+	// resourceID of the MSI that should be used (only relevant when using MSI, can not be used at the same time as ManagedIdentityClientID)
+	// +optional
+	ManagedIdentityResourceID string `json:"managedIdentityResourceID,omitempty"`
 }
```

Comment:
I did not recognize the acronym MSI and found this paragraph in https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview
> Managed identities for Azure resources is the new name for the service formerly known as Managed Service Identity (MSI).

So perhaps we should not be using MSI in the docs. 

Also, would it be simpler to validate and perhaps more future proof if we added these fields to a new struct. E.g.

```go
type ACMEIssuerDNS01ProviderAzureDNS struct {
  //+optional
  ManagedIdentity *AzureManagedIdentity `json: managedIdentity,optional`
}

type AzureManagedIdentity struct {
    ClientID string `json:"clientID,omitempty"`
    ResourceID string `json:"resourceID,omitempty"`
    ... # Room for other ManagedIdentity fields in the future.
}
```

I don't want to complicate this feature too much but just as a talking point, and imagining we could start from scratch, what would be the most intuitive layout for all these `ACMEIssuerDNS01ProviderAzureDNS` fields in your opinion?

Reply:
I've moved it to a nested struct, please let me know what you think